horace_core/sqw/file_io/@faccess_sqw_v3/get_header.m

```diff
@@ -1,96 +1,79 @@
 
-[headers,pos] = get_header@sqw_binfile_common(obj,varargin{:});
+%
+n_runs = exp_info.n_runs;
+if no_instr && no_sampl
+    return;
+end
+% only one experiment
 if get_all
     instr = obj.get_instrument('-all');
-    sampl = obj.get_sample('-all');
+    main_sampl = obj.get_sample('-all');
 else
     instr = obj.get_instrument(varargin{:});
-    sampl = obj.get_sample(varargin{:});
+    main_sampl = obj.get_sample(varargin{:});
 end
-
-n_files = numel(headers);
-
-if isstruct(headers)
-    if isstruct(instr)
-        if isempty(instr) || isempty(fieldnames(instr))
-            instr = IX_null_inst();
-        else
-            instr = IX_inst(instr);
-        end
-    end
-    headers.instrument = instr;
-    if isstruct(sampl)
-        if isempty(sampl) || isempty(fieldnames(sampl))
-            sampl = IX_null_sample;
-        else
-            sampl = IX_sample(sampl);
+if ~isempty(main_sampl)
+    if numel(main_sampl) > 1
+        % nsampl needs to be equal to number of runs
+        if numel(main_sampl) ~= exp_info.n_runs
+            error('HORACE:file_io:runtime_error',...
+                'Multiple sample in footer contains %d runs and number of runs stored in header=%d',...
+                numel(main_sampl),exp_info.n_runs)
         end
+    else % we need to propagate the sample, stored in the footer to all headers
+        main_sampl = repmat(main_sampl,1,exp_info.n_runs);
     end
-    headers.sample = sampl;
 else
-    for i=1:n_files
-        if numel(instr)>1
-            if iscell(instr)
-                inst = instr{1};
-            else
-                inst = instr(i);
-            end
-        else
-            inst = instr;
-        end
-        if isstruct(inst)
-            if isempty(inst) || isempty(fieldnames(inst))
-                inst = IX_null_inst();
-            else
-                inst = IX_inst(inst);
-            end
-        end
-        headers{i}.instrument = inst;
+    main_sampl = exp_info.samples;
+end
 
-        if numel(sampl)>1
-            if iscell(sampl)
-                samp = sampl{1};
-            else
-                samp = sampl(i);
-            end
-        else
-            samp = sampl;
+if ~isempty(main_sampl(1)) && (isempty(main_sampl(1).alatt) || isempty(main_sampl(1).angdeg))
+    % some odd bug in old file formats? sample lattice is not stored with
+    % sample
+    for i=1:n_runs
+        header_sampl = exp_info.samples(i);
+        if isempty(main_sampl(i).alatt)
+            main_sampl(i).alatt = header_sampl.alatt;
         end
-        if isstruct(samp)
-            if isempty(samp) || isempty(fieldnames(samp))
-                samp = IX_null_sample();
-            else
-                samp = IX_sample(samp);
-            end
+        if isempty(main_sampl(i).angdeg)
+            main_sampl(i).angdeg = header_sampl.angdeg;
         end
-        headers{i}.sample = samp;
     end
 end
-
-
+%
+if ~any(isempty(instr))
```

Comment:
I wrote this in assumption that instrument build with empty constructor is empty. You have slightly different approach for this. I do not mind you implementing it, so this will change. 
In addition to that we do not probably want to create arrays of empty instruments (what's the point) this is aligned with your next job of building proper experiment class. 
Feel free to replace this with proper Experiment construction

Reply:
1. Please add this information as a %TODO comment.
2. I want to move away from describing minimum information objects as empty. The empty concept is different. If you want to detect min info classes fine but not via empty

Reply:
I think I've substantially rewrote  the function to make it much clearer